zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/RequestMapper.java

```diff
@@ -609,26 +609,37 @@ public static Either<ProblemDetail, BroadcastSignalRequest> toBroadcastSignalReq
   }
 
   public static Authentication getAuthentication() {
-    String authenticatedUsername = null;
+    final String authenticatedUsername;
+    final String authenticatedApplicationId;
     final Map<String, Object> claims = new HashMap<>();
-    final List<String> authenticatedRoleIds = new ArrayList<>();
-    final List<String> authenticatedTenantIds = new ArrayList<>();
     final List<String> authenticatedMappingIds = new ArrayList<>();
+    final var authenticationBuilder = new Builder();
 
     final var requestAuthentication = SecurityContextHolder.getContext().getAuthentication();
     if (requestAuthentication != null) {
       if (requestAuthentication.getPrincipal()
           instanceof final CamundaPrincipal authenticatedPrincipal) {
         final var authenticationContext = authenticatedPrincipal.getAuthenticationContext();
 
-        authenticatedRoleIds.addAll(
-            authenticationContext.roles().stream().map(RoleEntity::roleId).toList());
-
-        authenticatedTenantIds.addAll(
-            authenticationContext.tenants().stream().map(TenantDTO::tenantId).toList());
-
-        authenticatedUsername = authenticationContext.username();
-        claims.put(Authorization.AUTHORIZED_USERNAME, authenticationContext.username());
+        final List<String> authenticatedRoleIds =
+            new ArrayList<>(
```

Comment:
:question: Why wrap this and `authenticatedTenantIds` in a new `ArrayList`? as far as I can tell, it's not being modified again.

Reply:
I think this was just an oversight on my part with code completion, I've now changed this